types/node/vm.d.ts

```diff
@@ -79,13 +90,51 @@ declare module "vm" {
         };
     }
 
+    interface SourceTextModuleOptions {
+        url?: string;
+        context?: Context;
+        lineOffset?: number;
+        columnOffset?: number;
+        initializeImportMeta?(meta: ImportMeta, module: SourceTextModule): void;
+        importModuleDynamically?(specifier: string, module: SourceTextModule): Object | SourceTextModule;
+    }
+
+    interface SourceTextModuleEvaluateOptions {
+        timeout?: number;
+        breakOnSigint?: boolean;
+    }
+
+    type SourceTextModuleStatus = 'uninstantiated' | 'instantiating' | 'instantiated' | 'evaluating' | 'evaluated' | 'errored';
+    type SourceTextModuleLinkingStatus = 'unlinked' | 'linking' | 'linked' | 'errored';
+
     class Script {
+        cachedData?: Buffer;
+        cachedDataProduced?: boolean;
+        cachedDataRejected?: boolean;
         constructor(code: string, options?: ScriptOptions);
         runInContext(contextifiedSandbox: Context, options?: RunningScriptOptions): any;
         runInNewContext(sandbox?: Context, options?: RunningScriptOptions): any;
         runInThisContext(options?: RunningScriptOptions): any;
         createCachedData(): Buffer;
     }
+
+    /**
+     * This feature is only available with the --experimental-vm-modules command flag enabled.
+     * https://nodejs.org/api/vm.html#vm_class_vm_sourcetextmodule
+     */
+    class SourceTextModule {
```

Comment:
What produces this? I don't see a definition for `vm.createSourceTextModule`.

Reply:
An instance of this class is created using
```
const instance = new createSourceTextModule('source code...');
```

Reply:
Right... there's no definition for that here, though - the class is just an orphaned type.

Reply:
The `SourceTextModule` class *is* exported by `vm` when `--experimental-vm-modules` is enabled.

Reply:
The `SourceTextModule` class *is* exported by `vm` when `--experimental-vm-modules` is enabled.

NOTE: There is no `createSourceTextModule` function in Node 12, as far as I can see.